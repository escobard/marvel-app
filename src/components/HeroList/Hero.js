import React, { Component } from "react";
import PropTypes from "prop-types";
import { Card, CardHeader, CardText } from "material-ui/Card";
import FlatButton from "material-ui/FlatButton";
import Toggle from "material-ui/Toggle";
import { Link } from "react-router-dom";

import { img } from "../../utils"

export default class Hero extends Component {
  render() {
    let { name, description, urls, thumbnail: { path, extension } } = this.props
      .hero.name
      ? this.props.hero
      : { thumbnail: {} };

    return (
      <Link to={`/hero/${name}`} className="link" params={this.props}>
      <Card className="hero-card">
        <CardHeader
          className="header"
          title={name}
          avatar={img(path, "portrait_medium", extension)}
        />
        <CardText className="description">
          {description
            ? description.substring(0, 40) + "..."
            : "No description available"}
        </CardText>
        
          <FlatButton className="expand" label="Click to expand"/>
        
      </Card>
      </Link>
    );
  }
}

Hero.defaultProps = {
  hero: {}
};

Hero.propTypes = {
  heroes: PropTypes.object
};
