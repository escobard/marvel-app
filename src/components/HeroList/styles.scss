@import "../../styles/variables";

.hero-list {
	margin: $space-xlrg 0;
	.hero {
		margin-bottom: $space-xlrg;
		.hero-card {
			min-height: 210px;
		}
	}
}
@media screen and (max-width: $break-medium) {
	.hero {
		.header {
			div {
				width: 100% !important;
				float: left;
				padding-right: 0 !important;
				text-align: center;
			}
			img {
				display: block !important;
				float: none;
				margin: auto !important;
			}
		}
		.description {
			text-align: center;
		}
		.link {
			float: none;
			margin: auto;
			display: block;
			width: fit-content;
		}
	}
}