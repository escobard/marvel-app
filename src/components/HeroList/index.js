import React, { Component } from "react"
import PropTypes from 'prop-types';

import styles from "./styles.scss"

import Hero from "./Hero"

export default class HeroList extends Component{

	renderHeroes = (heroes) => {
		return heroes.map(( hero, index ) =>{
			let {name} = hero;
			return(
				<div key={index} className="hero col-md-4">
					<Hero hero={hero}/>
				</div>
				)
		})
	}

	render(){
		let { heroes } = this.props
		console.log('heroes', this.props.heroes)
		return(
				<div className="hero-list">
					<p>Hero list</p>
					{
						heroes.length > 1 
						?	this.renderHeroes(heroes) 
						:	<p>Loading...</p>
					}
				</div>
			)
	}
}

HeroList.defaultProps = {
  heroes: []
};

HeroList.propTypes = {
  heroes: PropTypes.array
};