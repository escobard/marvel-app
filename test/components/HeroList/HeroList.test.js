import React, { Component } from 'react'
import HeroList from '../../../src/components/HeroList';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('HeroList snapshot', () => {
	const initialState = {output:100}
	const mockStore = configureStore()
	let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<HeroList /> )  
    })

    it('+++ render the HeroList component', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});