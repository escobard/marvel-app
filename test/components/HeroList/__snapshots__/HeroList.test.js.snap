// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroList snapshot +++ render the HeroList component 1`] = `
<div
  className="hero-list"
>
  <p>
    Hero list
  </p>
  <p>
    Loading...
  </p>
</div>
`;
